kaepora/miniLock
# Firefox addon (closes #34)

(this commit is ahead of #177, please review this first)

This changeset allows to create a miniLock firefox add-on.

Here's what I did:
* [install add-on sdk](https://developer.mozilla.org/en-US/Add-ons/SDK/Tutorials/Installation)
 * note the  source bin/activate` step!
* modified miniLock to be used as a firefox add-on
 * the add-on sdk wants all-lowercase add-on names. so I had to change the name to minilock (lowercase "L") in package.json
 * jetpack add-on id in package.json
* added lib/main.js as a main entry point for the firefox add-on (pathname & file according to addon-sdk guidelines)
 * added lib/main.js to the linter
 added makefile rules for firefox add-on

To test the add-on:
* install the add-on sdk and activate the virtual environment (see MDN link from above)
* go to the minilock dir and enter `cfx run`. this opens a new firefox profile the minilock add-on enabled
* run `cfx xpi` to generate an XPI file


TODO:
* to get this into addons.mozilla.org, all source code should be present in non-minified form. Some files may be whitelisted if they are well known (e.g. jQuery), but considering that this project is AGPL, we should lead with a clear example.
